.github/workflows/build.yml

```diff
@@ -79,9 +82,7 @@ jobs:
       with:
         path: ~/.ccache
         key: ccache-ubuntu-build-${{ matrix.os }}-${{ github.sha }}
-        restore-keys: | 
-          ccache-ubuntu-build-${{ matrix.os }}
-          ccache-ubuntu-build-
```

Comment:
I think it won't make much sense to reuse cache of an another distro version, because this will only leave us with a larger cache to be stored at the end, without being able to reuse much of it in the first place.